### PR TITLE
Regression test for Impersonating Troubleshooter

### PR DESCRIPTION
**Pull request by labkey-tchad**
#### Rationale
The new `ImpersonatingTroubleshooterRole` needs testing.
`PermissionsEditor` automatically saves after several operations, which makes it difficult to test error conditions. It isn't widely used however (it duplicates a lot of functionality from the more populare `UIPermissionsHelper`).

#### Story
* https://github.com/LabKey/platform/issues/4866

#### Related Pull Requests
- https://github.com/LabKey/testAutomation/pull/1697
- https://github.com/LabKey/sampleManagement/pull/2201
- https://github.com/LabKey/labbook/pull/566

#### Changes
* Regression test for Impersonating Troubleshooter
* Additional regression tests for App Admin
* Update `PermissionsEditor` test page
* Make `PermissionsHelper.getRoles` static

### CODE REVIEW

File: src/org/labkey/test/tests/core/security/AppAdminRoleTest.java [hunk 1]
```diff
@@ -40,11 +42,18 @@ public class AppAdminRoleTest extends BaseWebDriverTest
 {
     private static final String APP_ADMIN = "appadmin@appadmin.test";
     private static final String USER = "user@appadmin.test";
+    private static final String ADMIN_GROUP = "Custom Admin Group";
```
Comment (labkey-adam) [inline src/org/labkey/test/tests/core/security/AppAdminRoleTest.java:45]: `SITE_ADMIN_GROUP` and `"Custom Site Admin Group"` for clarity?

File: src/org/labkey/test/util/PermissionsHelper.java [hunk 2]
```diff
@@ -33,42 +34,46 @@
...
 public abstract class PermissionsHelper
 {
-    private static final String permissionClassPkg = "org.labkey.api.security.roles.";
-    public String toRole(final String perm)
+    public static final String SITE_ADMIN_ROLE = "Site Administrator";
+    public static final String APP_ADMIN_ROLE = "Application Admin";
+    public static final String DEVELOPER_ROLE = "Platform Developer";
+    public static final String IMP_TROUBLESHOOTER_ROLE = "Impersonating Troubleshooter";
+
+    public static String toRole(final String name)
```
Comment (labkey-adam) [inline src/org/labkey/test/util/PermissionsHelper.java:42]: Should callers use the above constants to specify common roles instead of hard-coded strings?

**Review by labkey-adam — APPROVED** [review]
Looks good to me. Just a couple optional suggestions.

**Comment by labkey-adam** [comment]
Just FYI: I have a local change that disables the privileged roles on the permissions page for App Admins, which I think is a better experience. This test will need to be updated when (or before) that's merged.

**Review by labkey-danield — APPROVED** [review]

**Pull request merged**